src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -303,6 +327,19 @@ protected void shadowUpdated(Map<String, Object> desired, Map<String, Object> re
         }
     }
 
+    private void syncShadowDeploymentStatus(Map<String, Object> reported) {
+        // device does not have anything to report
+        if (lastDeploymentStatus.get() == null) {
+            return;
+        }
+        HashMap<String, Object> latestDeploymentStatus = populateReportedSectionOfShadow(lastDeploymentStatus.get());
+        if (!reported.get(ARN_FOR_STATUS_KEY).equals(latestDeploymentStatus.get(ARN_FOR_STATUS_KEY))
+                || !reported.get(STATUS_KEY).equals(latestDeploymentStatus.get(STATUS_KEY))) {
```

Comment:
Is calling `populateReportedSectionOfShadow()` necessary here? Seems redundant to me. Why not just directly compare `lastDeploymentStatus` and `reported`?

Reply:
Certain fields in the deployment status is optional like (DeploymentStatusDetails). If a deployment status does not have an optional field, we need to explicitly delete that fields in the shadow document. For this we include the filed name with its value set as null. 

So lastDeploymentStatus will have optional fields names with value as null, while reported may not have them. This prevents us from directly comparing the two.

Reply:
But you're only comparing `ARN_FOR_STATUS_KEY` and `STATUS_KEY`. What I meant is you can just read these two keys from `latestDeploymentStatus`.

Reply:
Updated likewise.